### ISSUE TITLE
Configuration cache - Intellij synchronization crashes

### ISSUE THREAD

**Issue by dcabasson** [gradle/gradle#15044]
When using configuration cache with a kotlin based gradle project in Intellij, the first synchronization works but the second (consuming the gradle cache) fails with the error:
```
No model of type 'DummyModel' is available in this build.
> No builders are available to build a model of type 'org.jetbrains.plugins.gradle.model.internal.DummyModel'.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
```

This can be reproduced by creating a new project with `gradle init` (gradle 6.7):
```
√ ps/gradle-xxx % gradle init
Starting a Gradle Daemon (subsequent builds will be faster)

Select type of project to generate:
  1: basic
  2: application
  3: library
  4: Gradle plugin
Enter selection (default: basic) [1..4] 2

Select implementation language:
  1: C++
  2: Groovy
  3: Java
  4: Kotlin
  5: Scala
  6: Swift
Enter selection (default: Java) [1..6] 3

Split functionality across multiple subprojects?:
  1: no - only one application project
  2: yes - application and library projects
Enter selection (default: no - only one application project) [1..2] 1

Select build script DSL:
  1: Groovy
  2: Kotlin
Enter selection (default: Groovy) [1..2] 2

Select test framework:
  1: JUnit 4
  2: TestNG
  3: Spock
  4: JUnit Jupiter
Enter selection (default: JUnit 4) [1..4] 1

Project name (default: gradle-xxx): 
Source package (default: gradle.xxx): 

> Task :init
Get more help with your project: https://docs.gradle.org/6.7/samples/sample_building_java_applications.html

BUILD SUCCESSFUL in 33s
2 actionable tasks: 2 executed

```

And just adding a `gradle.properties` file with the content:
```
org.gradle.unsafe.configuration-cache=true
org.gradle.unsafe.configuration-cache-problems=warn
org.gradle.unsafe.configuration-cache.max-problems=5000
```

Opening and resyncing the project using Intellij `2020.2.3` and kotlin plugin `1.4.10-release-IJ2020.2-1` reproduces the issue.

reproducer: [gradle-15044.zip](https://github.com/gradle/gradle/files/5465759/gradle-15044.zip)

Error in Intellij:
<img width="2241" alt="Screen Shot 2020-10-30 at 10 40 59" src="https://user-images.githubusercontent.com/1946465/97718615-8562d800-1a9c-11eb-81d0-7ad7e4859fdf.png">

**Comment by eskatos** [comment]
`DummyModel` is built by https://github.com/JetBrains/intellij-community/blob/master/plugins/gradle/tooling-extension-impl/src/org/jetbrains/plugins/gradle/tooling/internal/ExtraModelBuilder.java
which is registered by https://github.com/JetBrains/intellij-community/blob/master/plugins/gradle/tooling-extension-impl/src/org/jetbrains/plugins/gradle/tooling/internal/init/init.gradle#L64

The builder registration happens in a before evaluate callback which won't be executed when reusing the configuration cache. This explains the problem.

**Comment by eskatos** [comment]
This can be tested with >= `6.9-20201206230026+0000`

**Comment by efemoney** [comment]
Am I correct that this **did not** make it into 6.8.1? Because I have this behavior with 6.8.1, AGP 4.2.0-beta03. I'm okay to turn off configuration cache until a Gradle version with fix.

**Comment by eskatos** [comment]
@efemoney this is correct. The fix is available starting with `7.0-milestone-1`
